pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/QueryQuotaClusterIntegrationTest.java

```diff
@@ -338,32 +393,37 @@ private void verifyQuotaUpdate(double quotaQps) {
         try {
           double tableQuota = Double.parseDouble(sendGetRequest(
               "http://" + _brokerHostPort + "/debug/tables/queryQuota/" + getTableName() + "_OFFLINE"));
+          double logicalTableQuota = Double.parseDouble(sendGetRequest(
+              "http://" + _brokerHostPort + "/debug/tables/queryQuota/" + getLogicalTableName()));
           double dbQuota = Double.parseDouble(
               sendGetRequest("http://" + _brokerHostPort + "/debug/databases/queryQuota/default"));
           double appQuota = Double.parseDouble(
               sendGetRequest("http://" + _brokerHostPort + "/debug/applicationQuotas/default"));
 
           tableQuota = tableQuota == 0 ? Long.MAX_VALUE : tableQuota;
+          logicalTableQuota = logicalTableQuota == 0 ? Long.MAX_VALUE : logicalTableQuota;
           dbQuota = dbQuota == 0 ? Long.MAX_VALUE : dbQuota;
           appQuota = appQuota == 0 ? Long.MAX_VALUE : appQuota;
 
-          double actualQuota = Math.min(Math.min(tableQuota, dbQuota), appQuota);
+          double actualQuota = Math.min(Math.min(tableQuota, logicalTableQuota), Math.min(dbQuota, appQuota));
```

Comment:
The interleaving of either checking table or logical table quota is confusing. Is there an option of generalizing the function and passing the required info as parameters ?